services/repository/src/main/java/org/sagebionetworks/repo/web/ControllerAuditAspect.java

```diff
@@ -13,21 +15,37 @@
  */
 @Aspect
 public class ControllerAuditAspect {
-	
+
 	@Autowired
-	AccessIdListener listener;
+	AccessRecordListener listener;
 
 	/**
 	 * Look at all return objects ma
-	 * @param returnValue
+	 *
+	 * @param responseBody
 	 */
 	@AfterReturning(pointcut = "@annotation(org.springframework.web.bind.annotation.ResponseBody)", returning = "responseBody")
 	public void inspectResponseBody(Object responseBody) {
 		// extract the ID from the response body
-		String id = IdExtractorUtil.getObjectId(responseBody);
-		if(id != null){
+		String id = AccessRecordExtractorUtil.getObjectId(responseBody);
+		if (id != null) {
 			listener.setReturnObjectId(id);
 		}
 	}
-	
+
+	/**
+	 * Look at all request objects
+	 *
+	 * @param
+	 */
+
+	@Before("within(@org.springframework.stereotype.Controller *) && (args(.., @RequestBody requestBody))")

```

Comment:
I don't know how this can work, the "within" expression matches "any join point (method execution only in Spring AOP) within the package". You should probably use @within or @annotation instead.

Reply:
You do not need the joinPOint argument?